iron-core/src/main/java/io/axway/iron/core/internal/StoreManagerImpl.java

```diff
@@ -91,7 +94,7 @@
                 = connectableTransactions // to start only when connect is called and not with the first subscription
                 .share(); // because we subscribe the main consumer and the one in charge of notifying recovery done
 
-        Completable withTimeout = transactions.takeWhile(t -> !m_closed).timeout(1, SECONDS).ignoreElements();
+        Completable withTimeout = transactions.takeWhile(t -> !(state == State.CLOSE)).timeout(1, SECONDS).ignoreElements();
```

Comment:
state != State.CLOSE ?

Reply:
There was a rollback so we are back to !m_closed.